### PR TITLE
[S3] Add operation PutBucketNotificationConfiguration

### PR DESCRIPTION
**Pull request by sunkan**

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -214,6 +214,7 @@
                 "ObjectNotExists",
                 "PutObject",
                 "PutObjectAcl",
+                "PutBucketNotificationConfiguration",
```
Comment (jderusse) [inline manifest.json:217]: Should be sort alphabetically

File: src/Service/S3/src/Enum/Event.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace AsyncAws\S3\Enum;
+
+final class Event
+{
+    public const S_3_OBJECT_CREATED_ = 's3:ObjectCreated:*';
```
Comment (jderusse) [inline src/Service/S3/src/Enum/Event.php:7]: Oh... I don't like the name of this constant.
I'll fix something in the generator.

File: src/Service/S3/tests/Unit/Input/PutBucketNotificationConfigurationRequestTest.php [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                                ],
+                            ]),
+                        ]),
+                    ]),
+                ],
+                'QueueConfigurations' => [
+                    new QueueConfiguration([
+                        'Id' => 'change me',
+                        'QueueArn' => 'change me',
+                        'Events' => ['s3:ObjectCreated:*'],
```
Comment (jderusse) [inline src/Service/S3/tests/Unit/Input/PutBucketNotificationConfigurationRequestTest.php:43]: Please use constants in tests when possible

File: src/Service/S3/tests/Unit/Input/PutBucketNotificationConfigurationRequestTest.php [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    <TopicConfiguration>
+        <Id>change me</Id>
+        <Topic>change me</Topic>
+        <Event>s3:ObjectCreated:*</Event>
+        <Filter>
+            <S3Key><FilterRule><Name>prefix</Name><Value>change me</Value></FilterRule></S3Key>
+        </Filter>
+    </TopicConfiguration>
+    <QueueConfiguration>
+        <Id>change me</Id>
```
Comment (jderusse) [inline src/Service/S3/tests/Unit/Input/PutBucketNotificationConfigurationRequestTest.php:92]: I AWS documenation this Is loooks like an integer (https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketNotificationConfiguration.html)
Are you sure both string and int are allowed?

File: src/Service/S3/tests/Integration/S3ClientTest.php [hunk 5]
```diff
@@ -549,6 +557,60 @@ public function testListParts(): void
...
+                            ])],
+                        ]),
+                    ]),
+                ])],
+            ]),
+            'ExpectedBucketOwner' => 'change me',
+        ]);
+        $result = $client->PutBucketNotificationConfiguration($input);
+
+        self::assertTrue($result->resolve());
```
Comment (jderusse) [inline src/Service/S3/tests/Integration/S3ClientTest.php:611]: If the fake S3 docker image support this feature, please add more assertions ont the reponse

I suggest to replace "change me" values by more friedly names

**Review by jderusse — COMMENTED** [review]
Thank you for this PR.

I've just few comments and I should improve the generator to avoid constants `S_3_FOO`

File: src/Service/S3/src/Enum/Event.php (hunk 2 above)
Comment (jderusse) [reply src/Service/S3/src/Enum/Event.php:7]: https://github.com/async-aws/aws/pull/865

Comment (Nyholm) [reply src/Service/S3/src/Enum/Event.php:7]: Nice. @sunkan, could you rebase and rerun the generator please?

File: manifest.json (hunk 1 above)
Comment (Nyholm) [reply manifest.json]: This element here is added automatically. 

Sorting it is nice, but it should be done by the generator, right?

**Review by Nyholm — COMMENTED** [review]
Cool cool cool. 

It looks good. Just improve the testing a bit an Im happy =)

File: manifest.json (hunk 1 above)
Comment (jderusse) [reply manifest.json]: Yes, it should have been done automatically
https://github.com/async-aws/aws/blob/master/src/CodeGenerator/src/Command/GenerateCommand.php#L275-L278

My though is, @sunkan added it manually and ran `generate --all`

File: src/Service/S3/tests/Integration/S3ClientTest.php (hunk 5 above)
Comment (sunkan) [reply src/Service/S3/tests/Integration/S3ClientTest.php:623]: What I can see the response is just a simple 200 status code

File: src/Service/S3/tests/Unit/Input/PutBucketNotificationConfigurationRequestTest.php (hunk 4 above)
Comment (sunkan) [reply src/Service/S3/tests/Unit/Input/PutBucketNotificationConfigurationRequestTest.php]: In the examples they have both

```xml
<QueueConfiguration>
      <Id>1</Id>
</QueueConfiguration>
```

and

```xml
<CloudFunctionConfiguration>
    <Id>ObjectCreatedEvents</Id>
</CloudFunctionConfiguration>
```

File: src/Service/S3/tests/Integration/S3ClientTest.php (hunk 5 above)
Comment (jderusse) [reply src/Service/S3/tests/Integration/S3ClientTest.php:623]: Oh, indeed, I thought AWS returned an object.
It's ok then.

**Review by jderusse — APPROVED** [review]

**Review by Nyholm — APPROVED** [review]
Thank you

**Pull request merged**